File: Argus/Angular-Frontend/src/app/model/vehicle.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import {environment} from '../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class VehicleService {

  private baseUrl = `${environment.apiUrl}/springboot/api/vehicles`;

  constructor(private http: HttpClient) { }

  getVehicleById(id: number): Observable<any> {
    return this.http.get(`${this.baseUrl}/${id}`);
  }

  addVehicle(vehicle: any): Observable<any> {
    return this.http.post(`${this.baseUrl}`, vehicle);
  }

  updateVehicle(id: number, value: any): Observable<any> {
    return this.http.put(`${this.baseUrl}/${id}`, value);
  }

  deleteVehicle(id: number): Observable<any> {
    return this.http.delete(`${this.baseUrl}/${id}`, { responseType: 'text' });
  }

  getVehicleList(): Observable<any> {
    return this.http.get(`${this.baseUrl}`);
  }
}

File: Spring-Backend/src/main/java/com/springboot/SpringBackend/service/VehicleService.java
package com.springboot.SpringBackend.service;

import com.springboot.SpringBackend.model.Vehicle;

import java.util.List;
import java.util.Optional;

public interface VehicleService {
    List<Vehicle> getAllVehicles();

    Optional<Vehicle> getVehicleById(Long id);

    Vehicle createVehicle(Vehicle x);

    Vehicle updateVehicle(Vehicle x);

    void deleteVehicle(Vehicle x);

    void deleteAllVehicles();
}
